### DISCUSSION TITLE
Bypass Branch Protection for Certain Files

### DISCUSSION THREAD

**Discussion by LachlanNewman** [community/community#27397] [Pull Requests]
I was wondering is there a way to enable users to bypass branch protection on pull requests if only certain files have changed.
For example, we have an npm package that is constantly changed and a bunch of other repos that rely on it for builds. so every time we update the package we need to get like 10 PRs approved.

**Comment by sawyerbivens** [comment]
Not sure if I'm getting this right but under the protect branch page their should be options at the very bottom to Allow force pushes. And I think if you make sure "Include administrators" you'd probably be good

![image](https://user-images.githubusercontent.com/48366954/181398508-80dcb0af-1ccb-46f8-afa2-0629c64b976c.png)

**Reply by LachlanNewman** [reply]
Not quite what im saying

Say for example you have a branch protection policy that requires an approval before merging.

If the PR only has changes to `specific_file` they you could merge without requiring an approval.

**Doesn't Require Approval**
```mermaid 
gitGraph
       commit
       branch develop
       checkout develop
       commit id: "change specific_file only"
       checkout main
       merge develop
```

**Requires Approval**
```mermaid 
gitGraph
       commit
       branch develop
       checkout develop
       commit id: "change specific_file and another file"
       checkout main
       merge develop
```

**Reply by sawyerbivens** [reply]
Ah, well I don't believe there's a way to do that in Git/Github but I could be wrong

**Comment by gatewaycat** [comment]
Is there any way to do this in 2023?

Perhaps a github action or bot that **detects when a PR only touches approved files** and then **approves the PR**?
